src/main/java/communication/MessageReceiver.java

```diff
@@ -0,0 +1,99 @@
+package communication;
+
+import org.apache.log4j.Logger;
+import org.zeromq.ZMQ;
+import org.zeromq.ZMQ.Socket;
+
+import control.FBase;
+import crypto.CryptoProvider.EncryptionAlgorithm;
+import de.hasenburg.fbase.model.GetMissedMessageResponse;
+import exceptions.FBaseException;
+import model.JSONable;
+import model.config.NodeConfig;
+import model.data.DataIdentifier;
+import model.data.DataRecord;
+import model.data.MessageID;
+import model.messages.Command;
+import model.messages.Envelope;
+import model.messages.Message;
+
+public class MessageReceiver extends AbstractReceiver {
```

Comment:
Shouldn't this be a less generic name?

Reply:
We have a messagePort in the config, thats why I called it MessageReceiver. This receiver will also handle other types of messages in the future (such as the direct get request for a specific data record).

The receiver might also be used for synchronous communication, if needed later.

Reply:
How about DirectMessageReceiver then?